### DISCUSSION TITLE
网盘已使用容量/剩余容量信息

### DISCUSSION THREAD

**Discussion by gexgd0419** [AlistGo/alist#3147] [Q&A]
已挂载阿里云盘+百度网盘。

尝试用rclone挂载WebDAV，无论挂载哪个网盘的路径，都不能给出正确的已使用容量、剩余容量、总容量信息。

WebDAV的文件夹有如下属性：`quota-available-bytes`和`quota-used-bytes`，可用于向客户端返回容量配额的剩余量和已使用量信息。rclone以及其它WebDAV客户端也多采用这两个属性确定显示的可用空间量。alist似乎并不会返回此信息。

alist是否支持网盘容量用量信息的获取？

**Comment by y383688728** [comment]
有解决办法没？ windows下webdav始终显示的是C盘的大小，并不是WEBDAV的大小

**Reply by gexgd0419** [reply]
Windows自带的WebDAV挂载功能并不支持从WebDAV服务器获取并显示容量信息。显示的C盘大小是因为缓存位置在C盘上。

如果希望获得准确的WebDAV容量信息，需要使用支持该特性的第三方工具挂载，例如RaiDrive或者RClone。

不过也不是所有的WebDAV服务器都支持向客户端反馈容量大小。对于不提供容量信息的服务器，即使使用支持的客户端挂载，也只会显示无效的容量数值。

**Reply by ZhangTianrong** [reply]
如果只是为了显示总大小的话，可以在 rClone 挂载时通过 `--vfs-disk-space-total-size` 手动设置总大小，并用 `--vfs-used-is-size` 让 rClone 主动计算已用空间大小，但是也只是方便显示而已，WebDAV 服务端不主动实现 rClone about 所需的 attribute, 那么挂载的时候无法得知已占用的空间或者总文件数之类的信息，如果想要配合 rClone 的 Union backend 还是没有办法按照剩余空间或者已占用空间选择 remote.
